tasks/Node/src/modules/task-utils/envutil.ts

```diff
@@ -0,0 +1,63 @@
+import fs from 'fs';
+
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as deployutil from '../../modules/task-utils/deployutil';
+import * as resutil from '../../modules/task-utils/resourceutil';
+
+import { Aborter, AnonymousCredential, BlobURL, Models, StorageURL } from "@azure/storage-blob";
+import { ResourceManagementClient } from "@azure/arm-resources";
+
+export async function exportEnvironmentTemplate(exportEnvTemplateLocation: string, envTemplateLocation: string, envTemplateSasToken: string): Promise<any> {
+    if (!envTemplateLocation || !envTemplateSasToken) {
+        throw 'Missing Environment Location or Environment SAS Token as outputs variables.';
+    }
+
+    console.log('Exporting environment template.');
+
+    const templateFileName = 'azuredeploy.json';
+    const credential = new AnonymousCredential();
+    const pipeline = StorageURL.newPipeline(credential);
+
+    const blobUrl = new BlobURL(`${envTemplateLocation}/${templateFileName}${envTemplateSasToken}`, pipeline);
+    const response: Models.BlobDownloadResponse = await blobUrl.download(Aborter.none, 0);
+
+    if (response && response.readableStreamBody) {
+        tl.mkdirP(exportEnvTemplateLocation);
+        const templateFileLocation = `${exportEnvTemplateLocation}/${templateFileName}`;
+        const data = response.readableStreamBody.read().toString();
+        fs.writeFileSync(templateFileLocation, data, 'utf8');
+        console.log(`Environment template has been exported to file: ${templateFileLocation}`);
+    }
+
+    console.log('Environment template has been exported successfully.');
+}
+
+export async function setOutputVariables(armClient: ResourceManagementClient, envRgId: string, template: any): Promise<any> {
```

Comment:
I've seen this in a few places, but you want Promise<void> not Promise<any> for promises that don't return anything when resolved.

Reply:
Done.